src/chainstate/stacks/db/unconfirmed.rs

```diff
@@ -43,6 +43,26 @@ use vm::costs::ExecutionCost;
 
 pub type UnconfirmedTxMap = HashMap<Txid, (StacksTransaction, BlockHeaderHash, u16)>;
 
+pub struct ProcessedUnconfirmedState {
+    pub total_burns: u128,
+    pub total_fees: u128,
+    pub receipts: Vec<StacksTransactionReceipt>,
+    // tuple representing (index, sequence #), where the index corresponds to the starting index of
+    // receipts corresponding to that sequence number in the receipts vector
+    pub sequence_indices_for_receipts: Vec<(usize, u16)>,
```

Comment:
This seems like a very round-about way of just turning `receipts` into `Vec<Vec<StacksTransactionReceipt>>`.  Want to do that instead, and in doing so, reduce the complexity of `process_new_microblocks()` in `event_observer.rs`?